test/blackbox/run_blackbox_tests/test_cases_encrypted_storage.c

```diff
@@ -0,0 +1,764 @@
+
+/* Execute encrypted storage Test Case # 1 - Sanity Test*/
+static void test_case_encrypted_storage_01(void **state) {
+	execute_test(test_steps_encrypted_storage_01, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 1
+
+    Test Steps:
+    1. Open encrypted meshlink node instance by passing invalid arguments
+
+    Expected Result:
+    @ meshlink_open_encrypted @ API should return false and errno being set
+*/
+static bool test_steps_encrypted_storage_01(void) {
+
+	char key[] = "encryption_key";
+	size_t keylen = sizeof(key);
+	meshlink_handle_t *mesh;
+
+	// Opening encrypted mesh instance with NULL as confbase name
+
+	meshlink_set_log_cb(NULL, MESHLINK_DEBUG, logger);
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted(NULL, "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	// Opening encrypted mesh instance with string length as zero as confbase name
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("", "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	// Opening encrypted mesh instance with NULL as key name
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("encrypted_storage_conf.1", "nut", "test", DEV_CLASS_BACKBONE, NULL, keylen);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	// Opening encrypted mesh instance with key length as 0 bytes
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("encrypted_storage_conf.1", "nut", "test", DEV_CLASS_BACKBONE, key, 0);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 2 - Sanity Test */
+static void test_case_encrypted_storage_02(void **state) {
+	execute_test(test_steps_encrypted_storage_02, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 2
+
+    Test Steps:
+    1. Open an encrypted confbase instance of meshlink with a key and close it.
+    2. Open the meshlink node instance with the same confbase but with a wrong or another key
+        that's different from original.
+    3. Open the meshlink node instance with the same confbase but with a valid key with which
+        initially it's opened.
+
+    Expected Result:
+    On reopening meshlink node instance with valid key @ meshlink_open_encrypted @ API should
+    return valid mesh handle else it should return NULL pointer.
+*/
+static bool test_steps_encrypted_storage_02(void) {
+	meshlink_destroy("encrypted_conf");
+	meshlink_handle_t *mesh = meshlink_open_encrypted("encrypted_conf", "nut", "test", DEV_CLASS_BACKBONE, "right", 5);
+	assert_int_not_equal(mesh, NULL);
+	meshlink_close(mesh);
+
+	mesh = meshlink_open_encrypted("encrypted_conf", "nut", "test", DEV_CLASS_BACKBONE, "wrong", 5);
+	assert_int_equal(mesh, NULL);
+
+	if(!mesh) {
```

Comment:
But at this point, we have already asserted that `mesh` is `NULL`, so we don't need the if-statement. Also, since it's `NULL`, the call to `meshlink_close(mesh)` doesn't do anything, so it can be removed as well.